### PR TITLE
Fixup for #2744

### PR DESCRIPTION
**Pull request by cdmihai**
As @davidfowl pointed out, we should check for the existence of metadata. Added a check wherever the CLI is reading metadata.

### CODE REVIEW

File: src/Microsoft.DotNet.Compiler.Common/BindingRedirectGenerator.cs [hunk 1]
```diff
@@ -179,10 +179,17 @@ internal static AssemblyRedirect[] CollectRedirects(IEnumerable<AssemblyReferenc
...
+            var assemblyPath = arg.ResolvedPath;
+
             try
             {
-                using (var peReader = new PEReader(File.OpenRead(arg.ResolvedPath)))
+                using (var peReader = new PEReader(File.OpenRead(assemblyPath)))
                 {
+                    if (!peReader.HasMetadata)
+                    {
+                        throw new InvalidDataException($"{assemblyPath} does not contain managed metadata");
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Compiler.Common/BindingRedirectGenerator.cs]: Don't throw just skip it

File: src/Microsoft.DotNet.ProjectModel/Resolution/PackageDependencyProvider.cs [hunk 2]
```diff
@@ -113,6 +113,11 @@ private static IEnumerable<string> GetDependencies(string path)
         {
             using (var peReader = new PEReader(File.OpenRead(path)))
             {
+                if (!peReader.HasMetadata)
+                {
+                    throw new InvalidDataException($"{path} does not contain managed metadata");
```
Comment (davidfowl) [inline src/Microsoft.DotNet.ProjectModel/Resolution/PackageDependencyProvider.cs]: Just skip it

Comment (cdmihai) [inline src/Microsoft.DotNet.ProjectModel/Resolution/PackageDependencyProvider.cs]: Shouldn't there be at least a warning? If it's silently skipped then the app will most likely crash at runtime with weird messages with the user having no idea how to approach debugging.

---
In reply to: [61773268](https://github.com/dotnet/cli/pull/2814#discussion_r61773268) [](ancestors = 61773268)

**Comment by eerhardt** [comment]
@davidfowl @cdmihai - is there anything left to be done on this PR?

**Comment by eerhardt** [comment]
@davidfowl @cdmihai - is there anything left to be done on this PR?

If this is required for preview2, this will need to be ported to the preview2 branch.

**Comment by cdmihai** [comment]
@davidfowl 
Is this still relevant? If yes, should it throw or should it skip? If there's non-managed dlls there and we choose to skip them, what's the chance that runtime will fail? Or is it correct to actually skip them?

**Comment by davidfowl** [comment]
We should skip not throw.

**Comment by eerhardt** [comment]
@cdmihai - Can you fix this up? I'm closing out old PRs if they are no longer active.

**Comment by cdmihai** [comment]
Is next week fine? :)

**Comment by eerhardt** [comment]
Yup totally fine.  There's no rush, I just want to ensure open PRs are still actively being pursued.

**Comment by brthor** [comment]
Should we still be looking to get this PR in or should it be closed out? @cdmihai

**Comment by cdmihai** [comment]
Can be closed. No longer relevant.

**Pull request closed without merging**